### PR TITLE
wip/ji/profile-ui-improved

### PR DESCRIPTION
**Pull request by jacek-izykowski**

### CODE REVIEW

File: src/ui/connectionmanager.cc [hunk 1]
```diff
@@ -106,8 +106,8 @@ void ConnectionManager::locallyCreatedServerFinished(int exit_code,
 }
 
 void ConnectionManager::connectionDialogAccepted() {
-  if(connection_dialog_->runANewServer()) {
-    shut_down_server_on_close_ = connection_dialog_->shutDownServerOnClose();
+  local_server_ = connection_dialog_->runANewServer();
+  if(local_server_) {
```
Comment (mkow) [inline src/ui/connectionmanager.cc]: space after `if`

File: src/ui/connectionmanager.cc [hunk 2]
```diff
@@ -126,7 +126,8 @@ void ConnectionManager::startClient() {
       veles::util::version::string,
       "Veles UI",
       "Veles UI",
-      QByteArray::fromHex(connection_dialog_->authenticationKey().toUtf8()));
+      QByteArray::fromHex(connection_dialog_->authenticationKey().toUtf8()),
+      local_server_);
```
Comment (mkow) [inline src/ui/connectionmanager.cc]: Please indent this.
```C++
QByteArray::fromHex(connection_dialog_->authenticationKey().toUtf8()),
                    local_server_);
```

File: include/ui/connectionmanager.h [hunk 3]
```diff
@@ -81,7 +81,7 @@ class ConnectionManager : public QObject {
   QAction* kill_locally_created_server_action_;
   QProcess* server_process_;
   ConnectionDialog* connection_dialog_;
-  bool shut_down_server_on_close_;
+  bool local_server_;
```
Comment (mkow) [inline include/ui/connectionmanager.h]: It's bool, not a server object, so IMO it should be called `is_server_local_` or something similar.

File: include/client/networkclient.h [hunk 4]
```diff
@@ -61,7 +61,8 @@ class NetworkClient : public QObject {
       QString client_version,
       QString client_description,
       QString client_type,
-      QByteArray authentication_key);
+      QByteArray authentication_key,
```
Comment (mkow) [inline include/client/networkclient.h]: Should be `const QByteArray&`. While you're editing this method, please also fix the rest of arguments (same problem).

**Comment by jacek-izykowski** [comment]
@mkow You have actually commented commits from other branches.

File: src/ui/connectionmanager.cc (hunk 2 above)
Comment (adamiwaniuk) [inline src/ui/connectionmanager.cc]: ?

**Pull request merged**